frontend/src/metabase/query_builder/actions/querying.js

```diff
@@ -176,7 +177,7 @@ export const queryCompleted = (question, queryResults) => {
   return async (dispatch, getState) => {
     const [{ data }] = queryResults;
     const [{ data: prevData }] = getQueryResults(getState()) || [{}];
-    const originalQuestion = getOriginalQuestion(getState());
+    const originalQuestion = getOriginalQuestionWithParameterValues(getState());
```

Comment:
This is the critical line of the fix. Before, `originalQuestion` wouldn't have `parameterValues` extracted from the card's template tag defaults, but the `question` argument to `queryCompleted` did. So on line 183 below, `isDirty` would be `true` even if the query hadn't changed at all, and `maybeResetDisplay` would be called on line 194. After the change, originalQuestion includes the `parameterValues` from the card's template tag defaults, so `isDirty` would be false. Then `maybeResetDisplay` doesn't get called.